### ISSUE TITLE
Rustdoc does not warn when linking to a private associated item

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust#81981]
I tried this code:

```rust
//! Link to [S::f]

pub use inner::S;

mod inner {
    pub struct S;
    impl S {
        fn f() {}
    }
}
```

I expected to see this happen: Rustdoc warns that `S::f` is private, the same way it would warn if `S` were private.

Instead, this happened: Rustdoc gives no warning and generates no link.

### Meta

`rustdoc --version`: rustdoc 1.52.0-nightly (9778068cb 2021-02-07)
